ionoscloud/resource_autoscaling_group.go

```diff
@@ -0,0 +1,484 @@
+Notice that exactly one volume can be set to PRIMARY or all of them set to AUTO.`,
+									},
+									"bus": {
+										Optional:    true,
+										Type:        schema.TypeString,
+										Default:     autoscaling.BUSTYPE_VIRTIO,
+										Description: `The bus type of the volume. Default setting is 'VIRTIO'. The bus type 'IDE' is also supported.`,
+									},
+									"backup_unit_id": {
+										Type:        schema.TypeString,
+										Description: "The uuid of the Backup Unit that user has access to. The property is immutable and is only allowed to be set on a new volume creation. It is mandatory to provide either 'public image' or 'imageAlias' in conjunction with this property.",
+										Optional:    true,
+										Computed:    true,
+									},
+								}},
+						},
+					},
+				},
+			},
+			"datacenter_id": {
+				Type:             schema.TypeString,
+				Description:      "Unique identifier for the resource",
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsUUID),
+			},
+			"location": {
+				Type:        schema.TypeString,
+				Description: "Location of the data center.",
+				Computed:    true,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func resourceAutoscalingGroupCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(services.SdkBundle).AutoscalingClient
+
+	group, err := autoscalingService.GetAutoscalingGroupDataCreate(d)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occured getting data from the provided schema: %w", err))
+	}
+	if group.Properties != nil && group.Properties.Name != nil {
+		log.Printf("[DEBUG] autoscaling group data extracted: %+v", *group.Properties.Name)
+	}
+
+	autoscalingGroup, _, err := client.CreateGroup(ctx, *group)
+	if err != nil {
+		diags := diag.FromErr(fmt.Errorf("error creating autoscaling group: %w", err))
+		return diags
+	}
+
+	d.SetId(*autoscalingGroup.Id)
+	log.Printf("[INFO] autoscaling Group created. Id set to %s", *autoscalingGroup.Id)
+
+	if err := checkAction(ctx, client, d); err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+	if err := autoscalingService.SetAutoscalingGroupData(d, autoscalingGroup.Properties); err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceAutoscalingGroupRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+
+	client := meta.(services.SdkBundle).AutoscalingClient
+
+	group, apiResponse, err := client.GetGroup(ctx, d.Id(), 2)
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			log.Printf("[INFO] resource %s not found: %+v", d.Id(), err)
+			d.SetId("")
+			return nil
+		} else {
+			diags := diag.FromErr(fmt.Errorf("error while retrieving autoscaling group with id %v, %w", d.Id(), err))
```

Comment:
return `diag.FromErr` directly

Reply:
@cbeti-ionos solved in c9df22e0066ca69a7809de59cd124fe97fec9580